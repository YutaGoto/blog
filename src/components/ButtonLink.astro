---
import "./fontawesome"

interface Props {
  text: string
  linkTo: string
  iconName: string
  isLeft?: boolean
}

const { text, linkTo, iconName, isLeft } = Astro.props
---

<a
  class="mb-1 box-border border border-blue-400 p-2 text-blue-400 hover:bg-blue-400 hover:text-gray-900 dark:border-blue-700 dark:text-gray-100 dark:hover:bg-blue-700 dark:hover:text-gray-100 lg:mb-0"
  href={linkTo}
>
  {isLeft && <i class={`fas fa-${iconName}`} aria-hidden="true" />}
  <span>{text}</span>
  {!isLeft && <i class={`fas fa-${iconName}`} aria-hidden="true" />}
</a>
