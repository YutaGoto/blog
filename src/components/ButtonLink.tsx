import Link from "next/link"
import React from "react"
import "./fontawesome"

interface ButtonLinkProps {
  text: string
  linkTo: string
  iconName: string
  rel: string
  isLeft?: boolean
}

export const ButtonLink = ({
  text,
  linkTo,
  iconName,
  rel,
  isLeft = true,
}: ButtonLinkProps) => (
  <Link
    className="mb-1 box-border border border-blue-400 p-2 text-blue-400 hover:bg-blue-400 hover:text-gray-900 dark:border-blue-700 dark:text-gray-100 dark:hover:bg-blue-700 dark:hover:text-gray-100 lg:mb-0"
    href={linkTo}
    rel={rel}
  >
    {isLeft && <i className={`fas fa-${iconName}`} aria-hidden="true" />}
    <span>{text}</span>
    {!isLeft && <i className={`fas fa-${iconName}`} aria-hidden="true" />}
  </Link>
)
