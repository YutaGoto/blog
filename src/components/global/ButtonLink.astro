---
import Link from "@/components/global/Link.astro"

interface Props {
  text: string
  linkTo: string
  iconName: string
  isLeft?: boolean
}

const { text, linkTo, iconName, isLeft } = Astro.props
---

<Link
  className="mb-1 box-border border border-blue-400 p-2 text-blue-400 hover:bg-blue-400 hover:text-gray-900 lg:mb-0 @dark:border-blue-700 @dark:text-gray-100 @dark:hover:bg-blue-700 @dark:hover:text-gray-100"
  href={linkTo}
>
  {isLeft && <i class={`ri-${iconName}`} />}
  <span>{text}</span>
  {!isLeft && <i class={`ri-${iconName}`} />}
</Link>
