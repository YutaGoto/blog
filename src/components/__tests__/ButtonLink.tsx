import React from "react"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"
import { ButtonLink } from "../ButtonLink"

describe("ButtonLink", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ButtonLink text="test" iconName="angle-left" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("display the a element", () => {
    const { container } = render(
      <ButtonLink text="test" iconName="angle-left" />
    )
    expect(container.querySelector(`a`)).toBeInTheDocument()
  })
})
