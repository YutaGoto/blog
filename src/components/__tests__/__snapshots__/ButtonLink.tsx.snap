// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink renders correctly 1`] = `
<a
  className="mb-1 box-border border border-blue-400 p-2 text-blue-400 hover:bg-blue-400 hover:text-gray-900 dark:border-blue-700 dark:text-gray-100 dark:hover:bg-blue-700 dark:hover:text-gray-100 lg:mb-0"
  href=""
  rel=""
>
  <i
    aria-hidden="true"
    className="fas fa-angle-left"
  />
  <span>
    test
  </span>
</a>
`;
